### PR TITLE
chore(WalletApiHandler): change balance retrieval method to reliable source

### PR DESCRIPTION
**Pull request by lucasvazo**
web3.js installed -> yarn

### CODE REVIEW

File: src/renderer/src/api/WalletApiHandler.ts [hunk 1]
```diff
@@ -1,25 +1,47 @@
...
+              "type": "function"
+            };
+        const weth = new web3.eth.Contract([wethAbi], wethAddress);
+        const retrieveBalance = await weth.methods.balanceOf(address).call((error: Error, result: string) => {
+        if (error) {
+            console.log(error);
+            return "0.00000";
+        } else {
+            const balance = web3.utils.fromWei(result, 'ether');
+            return balance;
```
Comment (albertalef) [inline src/renderer/src/api/WalletApiHandler.ts:38]: Use the indentation pattern in this file (prettier or editorconfig)

File: src/renderer/src/api/WalletApiHandler.ts [hunk 2]
```diff
@@ -1,25 +1,47 @@
...
+                {
+                  "name": "",
+                  "type": "uint256"
+                }
+              ],
+              "payable": false,
+              "stateMutability": "view",
+              "type": "function"
+            };
+        const weth = new web3.eth.Contract([wethAbi], wethAddress);
```
Comment (albertalef) [inline src/renderer/src/api/WalletApiHandler.ts:31]: Separate this code in a method that return a weth contract receiving a wethAbi as a param. Put the infuraApiKey and the wethAddress in a class static constant, in future we will put this in the .env file.

File: resources/counter.txt
Comment (albertalef) [inline resources/counter.txt:1]: remove this file

**Review by albertalef — CHANGES_REQUESTED** [review]

**Comment by lucasvazo** [comment]
> This branch need to be a fix not a chore, right?

At first I thought it was a code bug, hence the branch name, but it was an issue with the old api.
I'll keep it to fix anyway, because of the branch name

File: src/renderer/src/api/WalletApiHandler.ts [hunk 3]
```diff
@@ -1,25 +1,54 @@
...
+                "type": "uint256"
+              }
+            ],
+            "payable": false,
+            "stateMutability": "view",
+            "type": "function"
+          };
+      const weth = new this.web3.eth.Contract([wethAbi], this.wethAddress);
+      return weth;
+    }
```
Comment (albertalef) [inline src/renderer/src/api/WalletApiHandler.ts:34]: Separate all this in other class, in future you will use a wethAbiFactory to make others requests to web3

File: src/renderer/src/api/WalletApiHandler.ts [hunk 4]
```diff
@@ -1,25 +1,54 @@
...
+    
+    
+    static async getBalance(address: string) {
+        const weth = await this.getWethContract()
+        const retrieveBalance = await weth.methods.balanceOf(address).call((error: Error, result: string) => {
+            if (error) {
+                console.log(error);
```
Comment (albertalef) [inline src/renderer/src/api/WalletApiHandler.ts:41]:
```suggestion
```

**Review by albertalef — CHANGES_REQUESTED** [review]

**Review by albertalef — APPROVED** [review]
Perfect!

**Pull request merged**